### ISSUE TITLE
Naked structs

### ISSUE THREAD

**Issue by AndyPook** [mono/CppSharp#760]
Is it possible to generate just the __Internal structs? ie without the wrapping classes.

I would do this myself but nothing is virtual and you don't seem to be able to replace the Generator.
I'm pretty sure all I want to do is override VisitClassDecl within CSharpSources. But without cut/pasting a lot of other stuff...

I hope I'm just missing some magic option :)

**Comment by tritao** [comment]
I don't think we any option for this, but would be an useful feature to add for sure.

Maybe a custom generator would be the best option, could be derived from CSharpSources with some overriden methods like you suggested.

You could also achieve this for now with a custom pass, and calling `CSharpSources.GenerateClassInternals` for every class.

**Comment by YaakovDavis** [comment]
@tritao 
You might find the following snippet useful.

```
public void Postprocess(Driver driver, ASTContext ctx)
{
    foreach (var m in driver.Context.Options.Modules)
    {
        var gen = driver.Generator;
        var gens = gen.Generate(m.Units.GetGenerated());
        foreach(CSharpSources g in gens)
        {
            var classes = g.TranslationUnits.SelectMany(u => u.Classes);
            foreach (var c in classes)
            {
                g.WriteLine("struct " + c.OriginalName);
                g.GenerateClassInternals(c);
            }

            var s = g.Generate().Replace("\r\npublic partial struct __Internal", "");
        }
    }
}
```

The above assumes you set the following:
```
options.GenerateSequentialLayout = true;
```

**Comment by giladlevi** [comment]
@YaakovDavis Just saw your snippet, and as far as I can see it's a nice hack that hard patches the code...

Does this play nicely with the rest of CppSharp?

If I understand correctly I can't just take this new generated naked struct and pass it along to generated API wrappers, right?

**Comment by YaakovDavis** [comment]
@giladlevi 
I'm not aware of a way you could do that, though I'm new to this lib so perhaps there is.

I tried modifying a fork, in order to accomplish this in a more streamlined way, but didn't manage compiling yet.
